pzsd_bot/cogs/reminders/reminders.py

```diff
@@ -0,0 +1,273 @@
+import asyncio
+import logging
+import re
+
+import pendulum
+from discord import Bot, Colour, Embed, Message
+from discord.ext.commands import Cog
+from sqlalchemy import delete, insert, select, update
+from sqlalchemy.engine import Row
+from sqlalchemy.sql.functions import count
+
+from pzsd_bot.db import Session
+from pzsd_bot.ext.scheduler import Scheduler
+from pzsd_bot.model import ReminderStatus, pzsd_user, reminder
+from pzsd_bot.settings import Emoji, ReminderSettings
+
+logger = logging.getLogger(__name__)
+
+REMINDER_PATTERN = re.compile(
+    r"remind me (?P<preposition>in|at|on) "
+    r"(?P<time>.+?)((?: every )(?P<interval>.+))? "
+    r"to (?P<reminder>.+)",
+    re.IGNORECASE | re.DOTALL,
+)
+DURATION_PATTERN = re.compile(
+    re.compile(
+        r"((?P<years>\d+?) ?(years|year|Y|y) ?)?"
+        r"((?P<months>\d+?) ?(months|month|m) ?)?"
+        r"((?P<weeks>\d+?) ?(weeks|week|W|w) ?)?"
+        r"((?P<days>\d+?) ?(days|day|D|d) ?)?"
+        r"((?P<hours>\d+?) ?(hours|hour|hrs?|H|h) ?)?"
+        r"((?P<minutes>\d+?) ?(minutes|minute|min|M) ?)?"
+        r"((?P<seconds>\d+?) ?(seconds|second|secs?|S|s))?"
+    )
+)
+
+MINIMUM_INTERVAL_FREQUENCY = pendulum.duration(days=1)
```

Comment:
why is 1 day minimum? I think 1 second should be minimum

Reply:
lmao are you insane? ain't no way I'm letting these trouble makers capable of having the bot send messages once per second

Reply:
To be clear, this constant is just the **frequency** allowed for a _recurring_ reminder. Maybe you thought this was the limit on how soon a reminder could come? If so, there is no limit on that. At least no limit from me. The only limit to that would be from the datetime library. I'm just preventing someone from making a reminder that is triggered every second, essentially making the bot spam.

Reply:
Yeah I realized that later but 1 day still seemed restrictive